### PR TITLE
Fix issues with setting sky to undefined

### LINKED ISSUE #4517: Can't disable all sky rendering when terrain is enabled
**maplibre-gl-js version**: 4.5.1

**browser**:  Vivaldi 6.8.3381.48

### Steps to Trigger Behavior

 1. Open demonstration
 2. Disable sky with checkbox (this calls `map.setSky(undefined)`)
 3. Notice the sky rendering turns off, but the horizon and fog colouring persists
 4. Disable terrain via the control in the top right, notice that horizon/fog colouring is now gone

### Link to Demonstration

https://jsbin.com/hibilakuno/edit?html,output

### Expected Behavior

The sky/horizon/fog rendering is disabled when terrain is enabled when calling `setSky(undefined)` or `sky: undefined` in a style spec

### Actual Behavior

Only sky rendering is disabled when terrain is enabled

### PR DESCRIPTION
**Pull request by HarelM**
## Launch Checklist
This is a fix to issues related to sky definition. 
- #4517

Fixes #4517

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [x] Link to related issues.
 - [x] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality.
 - [x] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

**Comment by HarelM** [comment]
I'm not sure what is the correct way to use animation for removing the sky and fog, but if someone would like to invest time in solving this they are welcome to do so...

**Comment by HarelM** [comment]
I've solved the animation issue in last commit.

File: src/style/style.ts [hunk 1]
```diff
@@ -1496,17 +1496,20 @@ export class Style extends Evented {
...
-                update = true;
-            }
-        }
-        for (const key in skyOptions) {
-            if (!deepEqual(skyOptions[key], sky[key])) {
-                update = true;
-                break;
+        if (!skyOptions && !sky) return;
+
+        if (skyOptions && !sky) {
```
Comment (sbachinin) [inline src/style/style.ts:1503]: I think a little shorter and more readable would be
```
if (skyOptions && sky) {
    for...
} else {
    update = true
}
```

**Review by sbachinin — APPROVED** [review]
Code looks ok except this one comment.
As for the overall design, it's difficult for me to judge

File: src/style/style.ts (hunk 1 above)
Comment (HarelM) [reply src/style/style.ts:1503]: But it's not enough - you need to check all 4 options...

Comment (HarelM) [reply src/style/style.ts:1503]: In any case, I think it's OK, if you feel strongly about it let me know...

Comment (sbachinin) [reply src/style/style.ts:1503]: I think it's not a big deal )

**Pull request merged**